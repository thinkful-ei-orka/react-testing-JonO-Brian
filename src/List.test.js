import React from 'react';
import List from './List';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('<List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        let propsVar = {header: 'test header', cards:[{id: 'a',title:'Test Card',content:'loren ipsom'}]}
        ReactDOM.render(<List {...propsVar}/>, div);

        ReactDOM.unmountComponentAtNode(div);
});
});

describe('<List />', () => {
    it('renders this UI as expected', () => {
        // Render the component, as JSON
        let propsVar = {header: 'test header', cards:[{id: 'a',title:'Test Card',content:'loren ipsom'}]}
        const tree = renderer.create(<List {...propsVar}/>).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});
// test('renders learn react link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
