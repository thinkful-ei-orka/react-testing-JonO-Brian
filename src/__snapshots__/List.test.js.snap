// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h3>
      test header
    </h3>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Test Card
      </h3>
      <p>
        loren ipsom
      </p>
    </div>
  </div>
</section>
`;
